internal/provider/resource_docker_registry_image_funcs.go

```diff
@@ -483,7 +482,16 @@ func deleteDockerRegistryImage(pushOpts internalPushImageOptions, registryWithPr
 	}
 
 	if username != "" {
-		req.SetBasicAuth(username, password)
+		if pushOpts.Registry != "ghcr.io" && !isECRRepositoryURL(pushOpts.Registry) && !isAzureCRRepositoryURL(pushOpts.Registry) && pushOpts.Registry != "gcr.io" {
+			req.SetBasicAuth(username, password)
+		} else {
+			if isECRRepositoryURL(pushOpts.Registry) {
+				password = normalizeECRPasswordForHTTPUsage(password)
+				req.Header.Add("Authorization", "Basic "+password)
+			} else {
+				req.Header.Add("Authorization", "Bearer "+base64.StdEncoding.EncodeToString([]byte(password)))
+			}
+		}
```

Comment:
I am unsure about this addition.
When I tested the different registries a few weeks ago, for the `DELETE` route the existing implementation, using basicAuth was sufficient.
I will test this again

Reply:
I was very specifically getting this error without these changes.
```
Got error deleting registry image: Bad credentials: 401 Unauthorized
```